src/visit_vtk/full/vtkVisItCutter.C

```diff
@@ -493,44 +496,64 @@ vtkVisItCutter::UnstructuredGridCutter()
   int cut=0;
 
   vtkUnstructuredGrid *grid = vtkUnstructuredGrid::SafeDownCast(input);
+#if LIB_VERSION_LE(VTK, 8,1,0)
   vtkIdType *cellArrayPtr = grid->GetCells()->GetPointer();
+#else
+  auto cellIter = vtk::TakeSmartPointer(grid->GetCells()->NewIterator());
+  vtkIdType numCellPts;
+  const vtkIdType *ptIds;
+#endif
   double *scalarArrayPtr = cutScalars->GetPointer(0);
   double tempScalar;
   cellScalars = cutScalars->NewInstance();
   cellScalars->SetNumberOfComponents(cutScalars->GetNumberOfComponents());
   cellScalars->Allocate(VTK_CELL_SIZE*cutScalars->GetNumberOfComponents());
- 
+
   if ( this->SortBy == VTK_SORT_BY_CELL )
     {
-    // Loop over all contour values.  Then for each contour value, 
+    // Loop over all contour values.  Then for each contour value,
     // loop over all cells.
     //
-    for (iter=0; iter < numContours && !abortExecute; iter++)
+    for (int iter=0; iter < numContours && !abortExecute; iter++)
       {
       // Loop over all cells; get scalar values for all cell points
       // and process each cell.
       //
-      for (cellId=0; cellId < numCells && !abortExecute; cellId++)
+#if LIB_VERSION_LE(VTK, 8,1,0)
+      for (vtkIdType cellId=0, cellArrayIt=0; cellId < numCells && !abortExecute; cellId++)
+#else
+      for (cellIter->GoToFirstCell(); !cellIter->IsDoneWithTraversal() && !abortExecute; cellIter->GoToNextCell())
+#endif
         {
         if ( !(++cut % progressInterval) )
           {
           vtkDebugMacro(<<"Cutting #" << cut);
           this->UpdateProgress ((double)cut/numCuts);
           abortExecute = this->GetAbortExecute();
           }
+#if LIB_VERSION_LE(VTK, 8,1,0)
+        vtkIdType numCellPts = cellArrayPtr[cellArrayIt++];
 
-        numCellPts = cellArrayPtr[cellArrayIt];
-        cellArrayIt++;
-        
         //find min and max values in scalar data
         range[0] = scalarArrayPtr[cellArrayPtr[cellArrayIt]];
-        range[1] = scalarArrayPtr[cellArrayPtr[cellArrayIt]];
-        cellArrayIt++;
-        
-        for (i = 1; i < numCellPts; i++)
+        range[1] = scalarArrayPtr[cellArrayPtr[cellArrayIt++]];
```

Comment:
There is an odd inconsistency between lines 538 and 539 and lines 550 and 551. In the first case, the iterator is incrimented. In the second, the zeroth entry is used in both. It took me a minute to conclude both loops wind up doing same thing. I think pulling original line 528 into the above assignment makes matters worse in this regard. The real reason `cellArrayIt++` is needed prior to entering loop at 542 is that `i` is the loop variable and is starting from `1`, not `0`.

Reply:
Sorry...this was really vague. Let me try to correct.
```suggestion
        // we post-inc. cellArrayIt here because succeeding loop begins @ 1, not 0
        range[1] = scalarArrayPtr[cellArrayPtr[cellArrayIt++]];
```

Reply:
I wasn't consistent about moving stand-alone 'cellArrayIt++' so I think I will just revert that part of this change.